core/extension/gdextension_interface.h

```diff
@@ -255,12 +255,20 @@ typedef struct {
 	GDExtensionVariantPtr *default_arguments;
 } GDExtensionMethodInfo;
 
+typedef struct {
+	GDExtensionStringPtr message;
+	GDExtensionStringNamePtr property_name;
+	int32_t severity;
+} GDExtensionConfigurationInfo;
+
 typedef const GDExtensionPropertyInfo *(*GDExtensionClassGetPropertyList)(GDExtensionClassInstancePtr p_instance, uint32_t *r_count);
 typedef void (*GDExtensionClassFreePropertyList)(GDExtensionClassInstancePtr p_instance, const GDExtensionPropertyInfo *p_list);
 typedef void (*GDExtensionClassFreePropertyList2)(GDExtensionClassInstancePtr p_instance, const GDExtensionPropertyInfo *p_list, uint32_t p_count);
 typedef GDExtensionBool (*GDExtensionClassPropertyCanRevert)(GDExtensionClassInstancePtr p_instance, GDExtensionConstStringNamePtr p_name);
 typedef GDExtensionBool (*GDExtensionClassPropertyGetRevert)(GDExtensionClassInstancePtr p_instance, GDExtensionConstStringNamePtr p_name, GDExtensionVariantPtr r_ret);
 typedef GDExtensionBool (*GDExtensionClassValidateProperty)(GDExtensionClassInstancePtr p_instance, GDExtensionPropertyInfo *p_property);
+typedef const GDExtensionConfigurationInfo *(*GDExtensionClassGetConfigurationInfo)(GDExtensionClassInstancePtr p_instance, uint32_t *r_count);
+typedef void (*GDExtensionClassFreeConfigurationInfo)(GDExtensionClassInstancePtr p_instance, const GDExtensionConfigurationInfo *p_list, uint32_t p_count);
```

Comment:
Are these necessary as dedicated C functions, or could they simply be virtual functions on `Object` or `Node`?

Reply:
I copied the approach of `_get_property_list` since it seems most similar to what I'm doing. I don't really know if all of the GDExtension boilerplate I carried over is *needed* for users to be able to implement their own configuration info. But it seems like Object can't (or doesn't) use GDVIRTUAL, but is instead extending the GDCLASS macro etc.

> or `Node`

Inspector can show any Object, which is done e.g. for Editor/ProjectSettings, export options, some tilemap stuff. So the idea here is that we could later add configuration info to more places than just Nodes/Resources.